### PR TITLE
Don't refresh externals deps when pinned to a commit we already have

### PR DESCRIPTION
**Pull request by tspacek**
First draft at addressing https://github.com/CocoaPods/CocoaPods/issues/2949

### CODE REVIEW

**Comment by kylef** [comment]
@tspacek Would you be able to add specs for this?

File: lib/cocoapods/installer/analyzer.rb [hunk 1]
```diff
@@ -304,6 +304,7 @@ def dependencies_to_fetch
 
           if update_mode == :all
             deps_to_fetch = deps_with_external_source
+            deps_to_fetch -= deps_to_fetch.select{ |dep| !dep.external_source[:commit].nil? && !checkout_requires_update?(dep) }
           else
             deps_to_fetch = deps_with_external_source.select { |dep| pods_to_fetch.include?(dep.name) }
```
Comment (segiddins) [inline lib/cocoapods/installer/analyzer.rb]: the logic also needs to be duplicated here

**Comment by tspacek** [comment]
Yup I'll add some specs

**Comment by orta** [comment]
^ ping

**Comment by orta** [comment]
( don't want to be _too_ annoying, but would rather this didn't become a hung PR :) )

**Comment by segiddins** [comment]
I think this won't really be necessary once we have pod caching in place.

**Comment by kylef** [comment]
Thanks for your pull request and contribution @tspacek, I'm going to close this since it will be no longer necessary with #3236.

**Pull request closed without merging**